util/file_checksum_helper.h

```diff
@@ -16,24 +16,19 @@ namespace ROCKSDB_NAMESPACE {
 
 // This is the class to generate the file checksum based on Crc32. It
 // will be used as the default checksum method for SST file checksum
-class FileChecksumFuncCrc32c : public FileChecksumFunc {
+class FileChecksumGenCrc32c : public FileChecksumGenerator {
  public:
-  std::string Extend(const std::string& init_checksum, const char* data,
-                     size_t n) override {
-    assert(data != nullptr);
-    uint32_t checksum_value = StringToUint32(init_checksum);
-    return Uint32ToString(crc32c::Extend(checksum_value, data, n));
+  FileChecksumGenCrc32c(const FileChecksumGenContext& /*context*/) {
+    checksum_ = 0;
   }
 
-  std::string Value(const char* data, size_t n) override {
-    assert(data != nullptr);
-    return Uint32ToString(crc32c::Value(data, n));
+  void Update(const char* data, size_t n) override {
+    checksum_ = crc32c::Extend(checksum_, data, n);
   }
 
-  std::string ProcessChecksum(const std::string& checksum) override {
-    uint32_t checksum_value = StringToUint32(checksum);
-    return Uint32ToString(crc32c::Mask(checksum_value));
-  }
+  void Finalize() override { checksum_str_ = Uint32ToString(checksum_); }
```

Comment:
We should avoid/disallow calling `Finalize` multiple times. Currently it's possible due to the implementation of `GetFileChecksum()`.